pkg/mco/kubeletconfig.go

```diff
@@ -0,0 +1,268 @@
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating KubeletConfig %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.KubeletConfigs().Create(
+			context.Background(), builder.Definition, metav1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Delete removes the kubeletconfig.
+func (builder *Builder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the kubeletconfig object %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return fmt.Errorf("kubeletconfig cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.KubeletConfigs().Delete(
+		context.TODO(), builder.Object.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return fmt.Errorf("cannot delete kubeletconfig: %w", err)
+	}
+
+	builder.Object = nil
+
+	return err
+}
+
+// Exists checks whether the given kubeletconfig exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if the kubeletconfig object %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.KubeletConfigs().Get(
+		context.Background(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithMCPoolSelector redefines kubeletconfig definition with the given machineConfigPoolSelector field.
+func (builder *Builder) WithMCPoolSelector(key, value string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Labeling the kubeletconfig %s with %s=%s", builder.Definition.Name, key, value)
+
+	if key == "" {
+		glog.V(100).Infof("The key can't be empty")
+
+		builder.errorMsg = "'key' cannot be empty"
+
+		return builder
+	}
+
+	if builder.Definition.Spec.MachineConfigPoolSelector == nil {
+		builder.Definition.Spec.MachineConfigPoolSelector = &metav1.LabelSelector{}
+	}
+
+	if builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels == nil {
+		builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels = map[string]string{}
+	}
+
+	builder.Definition.Spec.MachineConfigPoolSelector.MatchLabels[key] = value
+
+	return builder
+}
+
+// WithSystemReserved redefines kubeletconfig definition with the given systemreserved fields.
+func (builder *Builder) WithSystemReserved(cpu, memory string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting up %s cpu and %s memory to the %s kubeletconfig",
+		cpu, memory, builder.Definition.Name)
+
+	if cpu == "" {
+		glog.V(100).Infof("The cpu can't be empty")
+
+		builder.errorMsg = "'cpu' cannot be empty"
+
+		return builder
+	}
+
+	if memory == "" {
+		glog.V(100).Infof("The memory can't be empty")
+
+		builder.errorMsg = "'memory' cannot be empty"
+
+		return builder
+	}
+
```

Comment:
Instead of returning builder twice above, just check if errorMsg != "" here and return if so

Reply:
sure